File: debug_jobtype_filtering.dart
// Debug script to check job type filtering issue
// Run this in Dart to investigate the kitchen staff / bartender checklist visibility issue

import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:firebase_core/firebase_core.dart';

Future<void> main() async {
  await Firebase.initializeApp();
  final firestore = FirebaseFirestore.instance;

  print('🔍 DEBUGGING JOB TYPE FILTERING ISSUE');
  print('===================================\n');

  // Test with the main organization
  final orgId = 'vnE0olvi1Tswjtdb19MI';

  try {
    // 1. Check checklist templates and their job types
    print('📋 CHECKING CHECKLIST TEMPLATES:');
    final templatesSnap =
        await firestore.collection('organizations').doc(orgId).collection('checklist_templates').get();

    for (final doc in templatesSnap.docs) {
      final data = doc.data();
      final name = data['name'] ?? 'Unknown';
      final jobTypes = data['jobTypes'] ?? data['jobType'];

      print('  Template: "$name" (${doc.id})');
      print('    Job Types: $jobTypes (${jobTypes.runtimeType})');

      if (name.toLowerCase().contains('kitchen') || name.toLowerCase().contains('bar')) {
        print('    ⚠️  RELEVANT TEMPLATE: $name has jobTypes = $jobTypes');
      }
    }

    // 2. Check a sample user's job types
    print('\n👤 CHECKING SAMPLE USERS:');
    final usersSnap = await firestore.collection('users').where('organizationId', isEqualTo: orgId).limit(5).get();

    for (final doc in usersSnap.docs) {
      final data = doc.data();
      final email = data['email'] ?? 'unknown';
      final jobTypes = data['jobTypes'] ?? data['jobType'];
      final userRole = data['userRole'] ?? 0;

      print('  User: $email (role: $userRole)');
      print('    Job Types: $jobTypes (${jobTypes.runtimeType})');
    }

    // 3. Check daily checklists and their job types
    print('\n📅 CHECKING DAILY CHECKLISTS:');
    final today = DateTime.now();
    final todayString =
        '${today.year.toString().padLeft(4, '0')}-${today.month.toString().padLeft(2, '0')}-${today.day.toString().padLeft(2, '0')}';

    final locationsSnap = await firestore.collection('organizations').doc(orgId).collection('locations').get();

    for (final locDoc in locationsSnap.docs) {
      final locationId = locDoc.id;
      final locationName = locDoc.data()['name'] ?? 'Unknown Location';

      print('  Location: $locationName ($locationId)');

      final checklistsSnap =
          await locDoc.reference.collection('daily_checklists').where('date', isEqualTo: todayString).get();

      for (final checklistDoc in checklistsSnap.docs) {
        final data = checklistDoc.data();
        final templateName = data['templateName'] ?? 'Unknown';
        final jobTypes = data['jobTypes'] ?? data['jobType'];

        print('    Checklist: "$templateName" (${checklistDoc.id})');
        print('      Job Types: $jobTypes (${jobTypes.runtimeType})');

        if (templateName.toLowerCase().contains('kitchen') || templateName.toLowerCase().contains('bar')) {
          print('      🚨 ISSUE: "$templateName" should be filtered by job type but has: $jobTypes');
        }
      }
    }

    print('\n✅ Debug complete. Check the output above for job type mismatches.');
  } catch (e) {
    print('❌ Error during debug: $e');
  }
}

File: debug_jobtype_filtering.js
const { initializeApp } = require('firebase/app');
const { getFirestore, collection, getDocs, query, where } = require('firebase/firestore');

// Initialize Firebase (use your config)
const firebaseConfig = {
  // Add your Firebase config here if needed for direct access
  // For now, we'll assume admin SDK is configured
};

const admin = require('firebase-admin');

// Initialize admin SDK if not already done
if (!admin.apps.length) {
  admin.initializeApp();
}

const db = admin.firestore();

async function debugJobTypeFiltering() {
  console.log('🔍 Debugging Job Type Filtering...\n');

  try {
    // 1. Check ChecklistTemplates - what job types do they have?
    console.log('📋 === CHECKLIST TEMPLATES ===');
    const templatesSnapshot = await db.collection('checklistTemplates').get();
    
    if (!templatesSnapshot.empty) {
      templatesSnapshot.forEach(doc => {
        const data = doc.data();
        console.log(`Template ID: ${doc.id}`);
        console.log(`  Name: ${data.name || 'N/A'}`);
        console.log(`  JobTypes: ${JSON.stringify(data.jobTypes || data.jobType || 'NONE')}`);
        console.log(`  Description: ${data.description || 'N/A'}`);
        console.log('---');
      });
    } else {
      console.log('❌ No checklist templates found');
    }

    // 2. Check actual DailyChecklists - what job types do they have?
    console.log('\n📋 === DAILY CHECKLISTS ===');
    const checklistsSnapshot = await db.collectionGroup('dailyChecklists').limit(10).get();
    
    if (!checklistsSnapshot.empty) {
      checklistsSnapshot.forEach(doc => {
        const data = doc.data();
        console.log(`Checklist ID: ${doc.id}`);
        console.log(`  Name: ${data.name || 'N/A'}`);
        console.log(`  JobTypes: ${JSON.stringify(data.jobTypes || data.jobType || 'NONE')}`);
        console.log(`  TemplateId: ${data.templateId || 'N/A'}`);
        console.log(`  Date: ${data.date || 'N/A'}`);
        console.log(`  Path: ${doc.ref.path}`);
        console.log('---');
      });
    } else {
      console.log('❌ No daily checklists found');
    }

    // 3. Check Users - what job types do they have?
    console.log('\n👥 === USERS ===');
    const usersSnapshot = await db.collection('users').limit(10).get();
    
    if (!usersSnapshot.empty) {
      usersSnapshot.forEach(doc => {
        const data = doc.data();
        console.log(`User ID: ${doc.id}`);
        console.log(`  Name: ${data.firstName || 'N/A'} ${data.lastName || 'N/A'}`);
        console.log(`  Email: ${data.email || data.emailAddress || 'N/A'}`);
        console.log(`  Role: ${data.userRole || 'N/A'}`);
        console.log(`  JobTypes: ${JSON.stringify(data.jobTypes || data.jobType || 'NONE')}`);
        console.log('---');
      });
    } else {
      console.log('❌ No users found');
    }

    // 4. Check specific Kitchen Staff vs Bartender issue
    console.log('\n🔍 === KITCHEN VS BARTENDER ANALYSIS ===');
    
    // Find kitchen staff users
    const kitchenUsersSnapshot = await db.collection('users')
      .where('jobTypes', 'array-contains-any', ['Kitchen Staff', 'kitchen staff', 'Kitchen', 'kitchen'])
      .get();
    
    console.log(`Found ${kitchenUsersSnapshot.size} kitchen staff users:`);
    kitchenUsersSnapshot.forEach(doc => {
      const data = doc.data();
      console.log(`  - ${data.firstName} ${data.lastName}: ${JSON.stringify(data.jobTypes || data.jobType)}`);
    });

    // Find bartender templates/checklists
    const bartenderTemplatesSnapshot = await db.collection('checklistTemplates')
      .where('jobTypes', 'array-contains-any', ['Bartender', 'bartender', 'Bar'])
      .get();
    
    console.log(`\nFound ${bartenderTemplatesSnapshot.size} bartender templates:`);
    bartenderTemplatesSnapshot.forEach(doc => {
      const data = doc.data();
      console.log(`  - ${data.name}: ${JSON.stringify(data.jobTypes || data.jobType)}`);
    });

  } catch (error) {
    console.error('❌ Error debugging job type filtering:', error);
  }
}

// Run the debug
debugJobTypeFiltering().then(() => {
  console.log('\n✅ Debug complete');
  process.exit(0);
}).catch(console.error);